src/build/tenscript/build_phase.rs

```diff
@@ -171,35 +172,37 @@ impl BuildPhase {
         }
     }
 
-    pub fn init(&mut self, fabric: &mut Fabric) {
+    pub fn init(&mut self, fabric: &mut Fabric, brick_library: &dyn BrickLibrary) -> Result<(), TenscriptError> {
         let (buds, marks) =
-            Self::execute_node(fabric, Scratch, &self.root, vec![]);
+            Self::execute_node(fabric, Scratch, &self.root, vec![], brick_library)?;
         self.buds = buds;
         self.marks = marks;
+        Ok(())
     }
 
     pub fn is_growing(&self) -> bool {
         !self.buds.is_empty()
     }
 
-    pub fn growth_step(&mut self, fabric: &mut Fabric) {
+    pub fn growth_step(&mut self, fabric: &mut Fabric, brick_library: &dyn BrickLibrary) -> Result<(), TenscriptError> {
```

Comment:
You can make all the `&dyn BrickLibrary`s into `impl BrickLibrary`s instead, to save runtime cost but also it's more proper. You shouldn't use `dyn` unless you really can't know the concrete type at runtime.

Reply:
e.g. if you want to create a vec of objects that implement a trait but have distinct concrete types

Reply:
Yes, thanks!